js/views/baseVw.js

```diff
@@ -150,6 +150,51 @@ export default class baseVw extends View {
     super.delegateEvents();
   }
 
+  /**
+   * Returns a given jQuery Element or a locally cached version for optimization.
+   * NOTE: Ensure that child views call super.render() in their render function to clear the cache
+   *       since the DOM elements in the cache will then be stale references.
+   * @param {selector} string - The jQuery selector for the Element.
+   *
+   * @return {element} JQuery The element(s) found in the View's Dom or the View's Cache
+   */
+  getCachedElement(selector) {
+    let element;
+
+    // Ensure we have our cached elements map.
+    if (!this._cachedElementMap) {
+      this._cachedElementMap = new Map();
+    }
+
+    // If the cache has the element, we shall use it.
+    if (this._cachedElementMap.has(selector)) {
+      element = this._cachedElementMap.get(selector);
+    } else {
+      // The cache does not not have the element, therefore query with jQuery and cache it.
```

Comment:
One thing to keep an eye out for is that if the selector doesn't find an element, it will set that cache to an empty element. In theory, that shouldn't be a problem (why keep going to the DOM for an element that's not there), but I remember a situation where that wasn't ideal and I ended up switching our manual cache to only cache if it finds the element, otherwise the next time it would also go to the DOM for it.

I can't remember the specific use case for that. Maybe it was due to the fact that sometimes in render we need to do something asynchronously via setTimeout since some plugins need an element to be in the DOM before initialization... and maybe an element generated by the plugin wouldn't be there immediately, but would later. Not sure.

We could probably leave it as is for now, but just something to keep in the back of our minds.